packages/db/src/translate/granules.ts

```diff
@@ -179,7 +179,7 @@ export const translateApiGranuleToPostgresGranule = async (
     updated_at: new Date(dynamoRecord.updatedAt),
   };
 
-  return granuleRecord;
+  return <PostgresGranule>removeNilProperties(granuleRecord);
```

Comment:
I'm going to 👍🏻  this as I know downstream work is going to need the undefined values.... but it does raise the question of why set the defaults *at all* in this code. 

Second question - why is the type annotation needed here?    TS seems to know that the object with nils removed still conforms to PostgresGranule unless I'm missing something.

Reply:
Yeah, it's not necessary to cast the return to the PostgresGranule type. I was being explicit because this is how it's done in e.g. collections.ts but it's redundant.

To your first question, you're referring to setting properties by a conditional like 
```
processing_end_date_time: dynamoRecord.processingEndDateTime
      ? new Date(dynamoRecord.processingEndDateTime) : undefined,
```
?

I think the issue there is that `new Date(undefined)` will throw an "Invalid Date" error. Is that what you're referring to?